cengage/react-magma
# fix(MultiCombobox): Support for `isClearable` button for multi comboboxes.

Issue: #907

## What I did

**New:** 
- Add `isTypeahead` prop to allow `selectedItems` items that are not in `items`
- Add `isLoading` prop to `ItemsList` to show a loading state. This means that when using both `isTypeahead && isLoading`, the loading indicator only appears on the items list (instead of inside the input)

**Fixes:**
- Support for `isClearable` button for multi comboboxes.
- `isClearable` button supports `isInverse` for single and multi comboboxes.
- Storybook story for Combobox `multi` now controls the `labelPosition`

## Screenshots (if appropriate):
![chrome-capture-2022-9-21](https://user-images.githubusercontent.com/91160746/197233936-aef85519-ea56-4ac6-9950-9fa3ed6dca8c.gif)
![chrome-capture-2022-9-21 (1)](https://user-images.githubusercontent.com/91160746/197234179-f92c59ec-f494-43a5-ad7d-0c9f7bc1b660.gif)
![image](https://user-images.githubusercontent.com/91160746/197234370-1e92901f-d0fc-4cc6-a0bb-b5924fa5feb1.png)

**isClearable && isInverse**
_before_
![image](https://user-images.githubusercontent.com/91160746/197234840-f89da379-ae13-41e0-bbc1-6b1e18e23a99.png)

_after_
![image](https://user-images.githubusercontent.com/91160746/197234508-7c7207bd-4b05-485d-b75d-7946602b6c4d.png)


## Checklist 

## How to test

- Confirm that `isClearable` button behaves as expected for single and multi comboboxes.
- Confirm that `isClearable` button supports `isInverse` for single and multi comboboxes.
- Confirm that when:
--  isTypeahead && isLoading -> loading indicator is only visible in the list
--  isTypeahead && !isLoading -> no loading indicator
--  !isTypeahead && isLoading -> loading indicator is only visible in the input
--  !isTypeahead && !isLoading -> no loading indicator
